### PR TITLE
Remove deprecated href

### PR DESCRIPTION
**Pull request by mosfet80**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7172">pipeline status</a> for feature branch [PR_7172](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7172). Pipeline [585761167](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/585761167) was triggered at [3c422ac](https://github.com/FreeCAD/FreeCAD/pull/7172/commits/3c422ac35a6a6e50292cf11d88f6cb0c8490f257). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by chennes** [comment]
We deliberately included HREF for "backwards" compatibility with @realthunder's fork.

File: src/Mod/Spreadsheet/App/PropertySheet.cpp [hunk 1]
```diff
@@ -1749,15 +1749,13 @@ bool PropertySheet::isBindingPath(const ObjectIdentifier &path,
...
-                && comps[1].getName()!="BindHREF"
-                && comps[1].getName()!="BindHiddenRef")
+            || (comps[1].getName()!="Bind" && comps[1].getName()!="BindHiddenRef")
             || path.getProperty() != this)
     {
         return false;
     }
     if(href)
-        *href = (comps[1].getName()=="BindHREF" || comps[1].getName()=="BindHiddenRef");
+        *href = ( comps[1].getName()=="BindHiddenRef");
```
Comment (luzpaz) [inline src/Mod/Spreadsheet/App/PropertySheet.cpp:1758]: please remove whitespace after opening parathesis

**Review by luzpaz — CHANGES_REQUESTED** [review]

**Comment by luzpaz** [comment]
> We deliberately included HREF for "backwards" compatibility with @realthunder's fork.

@chennes is the code commented to reflect that fact (maybe we can add 'Deprecated but kept for backward-compatibility with Link branch') ?

**Comment by realthunder** [comment]
Both my branch the upstream support both `href` and `hiddenref`. But I'd suggest we keep `href` for backward compatible. If I understand correctly, the reason for replacing `href` with `hiddenref` is to not confuse user for the more popular usage as hyper link tag. So I think it would be enough to just make it clear to user about the preference of `hiddenref` over 'href'. I've added this as tool tip in my expression completer PR. Speaking of which, @chennes @donovaly any chance to merge that PR #3062? Shall I do a rebase again?

**Comment by donovaly** [comment]
> Shall I do a rebase again?

Good question. I see there 50 commits and 55 changed files. I assume this is why it is stalled. I think it is maybe better to focus for now on toponaming and once this is set up we need time anyway for people to test. In this time I hope there will be manpower for this PR.
However, as you can imagine, many neat PRs are always good but I know that this is difficult in this case.

File: src/Mod/Spreadsheet/App/PropertySheet.cpp (hunk 1 above)
Comment (mosfet80) [reply src/Mod/Spreadsheet/App/PropertySheet.cpp]: Done..

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7172">pipeline status</a> for feature branch [PR_7172](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7172). Pipeline [586667119](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/586667119) was triggered at [c8790d5](https://github.com/FreeCAD/FreeCAD/pull/7172/commits/c8790d5f2479daac977e212b13c64a6cfd3a3530). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
> But I'd suggest we keep href for backward compatible.

OK and since also others here agreed, I take this as decision. -> Closing the PR now.

**Pull request closed without merging**